### PR TITLE
update links to configs

### PR DESCRIPTION
**Pull request by ruslandoga**
### Changes

This PR updates self-hosted config links to https://github.com/plausible/community-edition/tree/v2.1.0

### Tests
- [x] This PR does not require tests

### Changelog
- [x] This PR does not make a user-facing change

### Documentation
- [x] This change does not need a documentation update

### Dark mode
- [x] This PR does not change the UI

### CODE REVIEW

File: config/runtime.exs [hunk 1]
```diff
@@ -250,7 +250,7 @@ disable_registration =
   |> String.to_existing_atom()
 
 if disable_registration not in [true, false, :invite_only] do
-  raise "DISABLE_REGISTRATION must be one of `true`, `false`, or `invite_only`. See https://plausible.io/docs/self-hosting-configuration#server"
+  raise "DISABLE_REGISTRATION must be one of `true`, `false`, or `invite_only`. See https://github.com/plausible/community-edition/tree/v2.1.0?tab=readme-ov-file#disable_registration"
```
Comment (zoldar) [inline config/runtime.exs:253]: Anchor link reference does not seem to play well with the anchor hidden under <details> :( It opens the page scrolled to the very top. Do you want to leave it that way or perhaps wouldn't it be better to do away with foldable details?

Comment (ruslandoga) [reply config/runtime.exs:253]: Right, I'll update the docs to be flat with no details, similar to https://github.com/netlify/gotrue?tab=readme-ov-file#configuration to make everything easily linkable. I'll mark the PR as draft until I do that.

Comment (ruslandoga) [reply config/runtime.exs:253]: Done: https://github.com/plausible/community-edition/tree/v2.1.0?tab=readme-ov-file#disable_registration

Comment (zoldar) [reply config/runtime.exs:253]: Awesome 🙌

**Review by zoldar — APPROVED** [review]

**Pull request merged**